src/sentry/middleware/proxy.py

```diff
@@ -111,6 +111,16 @@ def __init__(self):
             from django.core.exceptions import MiddlewareNotUsed
             raise MiddlewareNotUsed
 
+    def _remove_port_number(self, ip_address):
+        if ']' in ip_address:
+            # IPv6 address with brackets, possibly with a port number
+            assert '[' in ip_address
```

Comment:
Can we move this into the if-condition actually? I don't think this should ever fail.

Reply:
Of course.